### PR TITLE
feat: add EffectComponent hit test

### PR DESCRIPTION
**Pull request by wumaolinmaoan**
## Summary by CodeRabbit

- **New Features**
	- Introduced advanced hit testing capabilities with the addition of triangles, bounding boxes, and hit test parameters in the effects core.
	- Added a new `Gizmos` class for drawing line strips and lines in 3D scenes, enhancing visual debugging tools.
- **Enhancements**
	- Improved particle trail rendering by optimizing index data handling.
	- Enhanced transformation calculations in the `Transform` class for more accurate positioning and rotation.
	- Streamlined WebGL geometry and renderer initialization for better performance and reliability.
- **Refactor**
	- Cleaned up import statements and removed unused imports across various modules to improve code readability and maintainability.
- **Bug Fixes**
	- Adjusted resource management in the `Player` class to prevent object loss during scene loading.

### CODE REVIEW

File: packages/effects-core/src/components/effect-component.ts [hunk 1]
```diff
@@ -8,6 +8,12 @@ import { Geometry } from '../render';
 import type { Disposable } from '../utils';
 import { DestroyOptions, generateGUID } from '../utils';
 import { RendererComponent } from './renderer-component';
+import { glContext, type spec } from '..';
+import { trianglesFromRect } from '../math';
+import type { HitTestTriangleParams, BoundingBoxTriangle } from '../plugins';
+import { HitTestType } from '../plugins';
+import { Vector3 } from '@galacean/effects-math/es/core/vector3';
+import type { TriangleLike } from '@galacean/effects-math/es/core';
```
Comment (yiiqii) [inline packages/effects-core/src/components/effect-component.ts:16]: 不要引 core，会把所有的代码都打进来，用具体的某个文件

File: packages/effects-core/src/components/effect-component.ts [hunk 2]
```diff
@@ -8,6 +8,12 @@ import { Geometry } from '../render';
 import type { Disposable } from '../utils';
 import { DestroyOptions, generateGUID } from '../utils';
 import { RendererComponent } from './renderer-component';
+import { glContext, type spec } from '..';
```
Comment (yiiqii) [inline packages/effects-core/src/components/effect-component.ts:11]: 明确引用路径

File: packages/effects-core/src/transform.ts [hunk 3]
```diff
@@ -1,8 +1,8 @@
 import { Euler, Matrix4, Quaternion, Vector2, Vector3 } from '@galacean/effects-math/es/core/index';
 import type * as spec from '@galacean/effects-specification';
+import type { Engine } from '.';
```
Comment (yiiqii) [inline packages/effects-core/src/transform.ts:3]: 明确引用路径

**Review by yiiqii — CHANGES_REQUESTED** [review]

**Review by liuxi150 — DISMISSED** [review]

File: packages/effects-core/src/components/effect-component.ts [hunk 4]
```diff
@@ -109,12 +121,46 @@ export class EffectComponent extends RendererComponent implements Disposable {
...
+
+    if (area) {
+      return {
+        type: area.type,
+        triangles: area.area,
+      };
+    }
+  };
+
+  getBoundingBox (): BoundingBoxTriangle | void {
```
Comment (Sruimeng) [inline packages/effects-core/src/components/effect-component.ts:135]: 只给编辑器用的话，加下注释

**Review by Sruimeng — DISMISSED** [review]

**Review by yiiqii — APPROVED** [review]

**Pull request merged**